### PR TITLE
test: analyze and add scrubbing tests (EMR screenshots)

### PR DESCRIPTION
**Pull request by KrishPatel13**
**What kind of change does this PR introduce?**

- Initiate the analysis of Scrubbing involving EMR and PHI data.
- Add Potential PHI involving EMR images and their scrubbed version.

**Summary**

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

**Comment by KrishPatel13** [comment]
Original Image:
sample_emr_1

Scrubbed Image:
sample_emr_1_scrubbed


Code that was used for scrubbing:
```
def test_analyze_emr() -> None:
    """
    Test to analyze some EMR PII data
    """

    # warnings.filterwarnings("ignore", category=DeprecationWarning)
    i = 1
    # resources/sample_emr_{i}.png
    test_image_path = "resources/sample_emr_"
    test_image_path = test_image_path + str(i) + ".png"
    with open(test_image_path, "rb") as file:
        test_image_data = file.read()

    # Convert image data to PIL Image object
    test_image = Image.open(BytesIO(test_image_data))

    # Scrub the image
    scrubbed_image = scrub.scrub_image(test_image)

    # Save the scrubbed image data to a file
    scrubbed_image_path = test_image_path[:-4] + "_scrubbed.png"
    scrubbed_image.save(scrubbed_image_path)
     
    # Assert that resources has the scrubbed files
    assert os.path.exists("resources/sample_emr_1_scrubbed.png")
```
```
def scrub_image(
    image: Image, fill_color=config.SCRUB_FILL_COLOR
) -> Image:
    """
    Scrub the image of all PII/PHI using Presidio Image Redactor

    Args:
        image (PIL.Image): A PIL.Image object to be scrubbed

    Returns:
        PIL.Image: The scrubbed image with PII and PHI removed.
    """
    
    redacted_image = IMAGE_REDACTOR.redact(
        image, fill=fill_color, entities=SCRUBBING_ENTITIES
    )

    return redacted_image
```

**Comment by KrishPatel13** [comment]
Linked an Issue at Presidio Github: https://github.com/microsoft/presidio/issues/1101

For the above issue.

**Comment by KrishPatel13** [comment]
@abrichr

Looks like we can have some sort of customization in ImageRedaction by providing our another OCR besides Tesseract.

https://github.com/microsoft/presidio/blob/main/presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py#L20

**Comment by KrishPatel13** [comment]
Next Steps:

- [ ] Investigate Tesseract Standards for optimal results on an ideal Image
- [ ]add resize according To tesseract standards : 
https://stackoverflow.com/questions/9480013/image-processing-to-improve-tesseract-ocr-accuracy (eg. resizing, binarizing etc.)

**Comment by KrishPatel13** [comment]
I have asked the Cape team to help me analyze pii/phi in a png.
image

**Comment by KrishPatel13** [comment]
Idea is to use Cape's deidentify file api to get the results after analyzing a png file. Then run the run the same function again on the scrubbed png file (from scrub_image function) and compare the diffs.

If there are still PII/PHI i diffs then that means the emr image was not completely scrubbed by the scrub_image function and hence
 it will result in a failing test (what we want in this PR)

**Comment by KrishPatel13** [comment]
@abrichr any thoughts on this approach ?

**Comment by KrishPatel13** [comment]
Next up: 

- [ ] Run Tesseract on the Emr image on both original and redacted images.
- [ ] Feed the both text to cape and get the diffs.
- [ ] If diff still has the NER (Name Entity Recognizer) then, assert that the test fail.

**Comment by KrishPatel13** [comment]
@abrichr any thoughts on this approach ? 🤔 https://github.com/OpenAdaptAI/OpenAdapt/pull/328#issuecomment-1666131402

**Comment by KrishPatel13** [comment]
Todo : add aws comprehend

**Comment by abrichr** [comment]
It's a good approach.

One limitation is that this assumes Tesseract is always accurate, which we know is incorrect.

**Comment by KrishPatel13** [comment]
Tried to test Amazon Comprehend here: https://github.com/OpenAdaptAI/OpenAdapt/pull/418/commits/46597ae764dd69461e7df329cc4018f1c01dc37d

Lodged an issues at their github: https://github.com/awsdocs/aws-doc-sdk-examples/issues/5218

**Comment by KrishPatel13** [comment]
Results of Pytest:

```
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\test_other\OpenAdapt> pytest
============================================================================================ test session starts ============================================================================================
platform win32 -- Python 3.10.11, pytest-7.1.3, pluggy-1.2.0
rootdir: P:\OpenAdapt AI - MLDS AI\cloned_repo\test_other\OpenAdapt
plugins: anyio-3.7.1, Faker-19.2.0, cov-2.10.0
collected 26 items

tests\openadapt\test_crop.py .                                                                                                                                                                         [  3%]
tests\openadapt\test_events.py .......                                                                                                                                                                 [ 30%]
tests\openadapt\test_scrub.py F...............                                                                                                                                                         [ 92%]
tests\openadapt\test_summary.py ..                                                                                                                                                                     [100%]

================================================================================================= FAILURES ==================================================================================================
______________________________________________________________________________________________ test_emr_image _______________________________________________________________________________________________

    def test_emr_image() -> None:
        """Test to verify that an EMR scrubbed image does not contain any PII/PHI."""
        test_image_path = "assets/test_emr_image.png"
        test_image_path_300 = "assets/test_emr_image_300.png"
        scrubbed_image_path = test_image_path[:-4] + "_scrubbed.png"
    
        _enhance_image_for_ocr(test_image_path, test_image_path_300)
        original_image_text = pytesseract.image_to_string(Image.open(test_image_path_300))
        os.remove(test_image_path_300)
    
        # Preparing for Scrubbing
        with open(test_image_path, "rb") as file:
            test_image_data = file.read()
    
        # Convert image data to PIL Image object
        test_image = Image.open(BytesIO(test_image_data))
    
        # Scrub the image
        scrubbed_image = scrub.scrub_image(test_image)

        # Save the scrubbed image data to a file
        scrubbed_image.save(scrubbed_image_path)

        _enhance_image_for_ocr(scrubbed_image_path, test_image_path_300)
        scrubbed_image_text = pytesseract.image_to_string(Image.open(test_image_path_300))
        os.remove(test_image_path_300)
        test_image.close()
        os.remove(scrubbed_image_path)

        # Use Cape to detect PII/PHI in the scrubbed image ocr text
        resp = requests.post(  # pylint: disable=missing-timeout
            "https://api.capeprivacy.com/v1/privacy/deidentify/text",
            headers={
                "Authorization": (
                    f"Bearer {config.CAPE_API_KEY}"  # pylint: disable=no-member
                )
            },
            json={
                "content": scrubbed_image_text,
            },
        )

>       assert (
            resp.json().get("entities") == []
        )  # Empty list means no PII or PHI was found by Cape
E       AssertionError: assert [{'best_label...'Smith'}, ...] == []
E         Left contains 31 more items, first extra item: {'best_label': 'NAME_GIVEN', 'processed_text': 'NAME_GIVEN_1', 'text': 'le'}
E         Use -v to get more diff

tests\openadapt\test_scrub.py:109: AssertionError
------------------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------------------- 
Token indices sequence length is longer than the specified maximum sequence length for this model (521 > 512). Running this sequence through the model will result in indexing errors
============================================================================================= warnings summary ============================================================================================== 
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pycountry\__init__.py:10
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pycountry\__init__.py:10: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/openadapt/test_scrub.py::test_emr_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pytesseract\pytesseract.py:382: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return LooseVersion(

tests/openadapt/test_scrub.py::test_emr_image
tests/openadapt/test_scrub.py::test_scrub_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\setuptools\_distutils\version.py:345: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================== short test summary info ========================================================================================== 
FAILED tests/openadapt/test_scrub.py::test_emr_image - AssertionError: assert [{'best_label...'Smith'}, ...] == []
================================================================================= 1 failed, 25 passed, 9 warnings in 18.64s ================================================================================= 
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\test_other\OpenAdapt> 
```

**Comment by KrishPatel13** [comment]
Note: A new version of Presidio Image redactor library was available, hence I update the poetry lick and toml files to accommodate it.

File: openadapt/config.py [hunk 1]
```diff
@@ -62,13 +62,13 @@
         # 'PHONE_NUMBER',
         # 'US_ITIN',
         # 'AU_ABN',
-        "DATE_TIME",
+        # "DATE_TIME",
```
Comment (KrishPatel13) [inline openadapt/config.py:65]: Uncomment DATE_TIME and URL entities as we no longer worry about the redaction of timestamps in visualization and also we want the most accurate scrubbing form presidio for testing.

File: tests/openadapt/test_scrub.py [hunk 2]
```diff
@@ -19,16 +21,100 @@ def _hex_to_rgb(hex_color: int) -> tuple[int, int, int]:
...
-    g = (hex_color >> 8) & 0xFF
-    r = hex_color & 0xFF
-    return r, g, b
+    black = (hex_color >> 16) & 0xFF
+    green = (hex_color >> 8) & 0xFF
+    red = hex_color & 0xFF
+    return red, green, black
+
+
+def _enhance_image_for_ocr(input_image_path: str, enhanced_image_path: str) -> None:
```
Comment (KrishPatel13) [inline tests/openadapt/test_scrub.py:30]: Enhance image properties for getting best results from TesseractOCR. See reference links

File: tests/openadapt/test_scrub.py [hunk 3]
```diff
@@ -19,16 +21,100 @@ def _hex_to_rgb(hex_color: int) -> tuple[int, int, int]:
...
+        cv2.ADAPTIVE_THRESH_MEAN_C,  # pylint: disable=no-member
+        cv2.THRESH_BINARY_INV,  # pylint: disable=no-member
+        23,
+        100,
+    )
+    bnt_image = cv2.bitwise_not(thr_image)  # pylint: disable=no-member
+    cv2.imwrite(enhanced_image_path, bnt_image)  # pylint: disable=no-member
+
+
+def test_emr_image() -> None:
```
Comment (KrishPatel13) [inline tests/openadapt/test_scrub.py:67]: Testing EMR IMage using Tesseract OCR and Cape Privacy

File: tests/openadapt/test_scrub.py [hunk 4]
```diff
@@ -62,7 +148,7 @@ def test_scrub_image() -> None:
 
     # Assert ~1.5% mask pixels compared to total pixels.
     assert (
-        round((mask_pixels / total_pixels), 3) == 0.015
+        round((mask_pixels / total_pixels), 3) == 0.022
```
Comment (KrishPatel13) [inline tests/openadapt/test_scrub.py:151]: Updated the values acc. to he changes in the latest update in the presidio image-redactor library

**Review by KrishPatel13 — COMMENTED** [review]
Self-reviewed! @abrichr Ready for review

**Comment by KrishPatel13** [comment]
To Do remove un-important entities from Cape response json.

**Comment by KrishPatel13** [comment]
Skipping test on Invalid CAPE_API_KEY:
```
tests\openadapt\test_scrub.py Token indices sequence length is longer than the specified maximum sequence length for this model (521 > 512). Running this sequence through the model will result in indexing errors
s...............
tests\openadapt\test_summary.py ..

============================================================================================= warnings summary ============================================================================================== 
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pycountry\__init__.py:10
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pycountry\__init__.py:10: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/openadapt/test_scrub.py::test_emr_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pytesseract\pytesseract.py:382: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return LooseVersion(

tests/openadapt/test_scrub.py::test_emr_image
tests/openadapt/test_scrub.py::test_scrub_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\setuptools\_distutils\version.py:345: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================ 25 passed, 1 skipped, 6 warnings in 16.88s ================================================================================= 
```


Failing test on Valid CAPE_API_KEY:
```
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\test_other\OpenAdapt> pytest   
============================================================================================ test session starts ============================================================================================
platform win32 -- Python 3.10.11, pytest-7.1.3, pluggy-1.2.0
rootdir: P:\OpenAdapt AI - MLDS AI\cloned_repo\test_other\OpenAdapt
plugins: anyio-3.7.1, Faker-19.2.0, cov-2.10.0
collected 26 items

tests\openadapt\test_crop.py .                                                                                                                                                                         [  3%]
tests\openadapt\test_events.py .......                                                                                                                                                                 [ 30%]
tests\openadapt\test_scrub.py F...............                                                                                                                                                         [ 92%]
tests\openadapt\test_summary.py ..                                                                                                                                                                     [100%] 

================================================================================================= FAILURES ================================================================================================== 
______________________________________________________________________________________________ test_emr_image _______________________________________________________________________________________________ 

    def test_emr_image() -> None:
        """Test to verify that an EMR scrubbed image does not contain any PII/PHI."""
        test_image_path = "assets/test_emr_image.png"
        test_image_path_300 = "assets/test_emr_image_300.png"
        scrubbed_image_path = test_image_path[:-4] + "_scrubbed.png"

        _enhance_image_for_ocr(test_image_path, test_image_path_300)
        # original_image_text = (
        #     pytesseract.image_to_string(Image.open(test_image_path_300))
        # )
        os.remove(test_image_path_300)

        # Preparing for Scrubbing
        with open(test_image_path, "rb") as file:
            test_image_data = file.read()

        # Convert image data to PIL Image object
        test_image = Image.open(BytesIO(test_image_data))

        # Scrub the image
        scrubbed_image = scrub.scrub_image(test_image)

        # Save the scrubbed image data to a file
        scrubbed_image.save(scrubbed_image_path)

        _enhance_image_for_ocr(scrubbed_image_path, test_image_path_300)
        scrubbed_image_text = pytesseract.image_to_string(Image.open(test_image_path_300))
        os.remove(test_image_path_300)
        test_image.close()
        os.remove(scrubbed_image_path)

        # try:
        # Use Cape to detect PII/PHI in the scrubbed image ocr text
        resp = requests.post(  # pylint: disable=missing-timeout
            "https://api.capeprivacy.com/v1/privacy/deidentify/text",
            headers={
                "Authorization": (
                    f"Bearer {config.CAPE_API_KEY}"  # pylint: disable=no-member
                )
            },
            json={
                "content": scrubbed_image_text,
            },
        )
        cape_response = resp.json()

        if (
            "entities" not in cape_response.keys()
            or "content" not in cape_response.keys()
            or cape_response is None
        ):
            pytest.skip(reason="Cape failed to return entities and content")

        detect_entities = cape_response.get("entities")
        important_entities = [
            "NAME",
            "NAME_GIVEN",
            "NAME_FAMILY",
            "TIME",
            "DATE",
            "PHONE_NUMBER",
            "DOB",
            "EMAIL_ADDRESS",
            "CREDIT_CARD",
            "LOCATION",
            "DRIVER_LICENSE",
            "DATE_INTERVAL",
            "SSN",
        ]
        filtered_entities = []

        for entity in detect_entities:
            label = entity.get("best_label")
            if label in important_entities and label not in filtered_entities:
                filtered_entities.append(label)

        logger.debug(filtered_entities)

>       assert (
            not filtered_entities
        )  # Empty list means no important PII or PHI was found by Cape
E       AssertionError: assert not ['NAME_GIVEN', 'NAME_FAMILY', 'PHONE_NUMBER']

tests\openadapt\test_scrub.py:147: AssertionError
------------------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------------------- 
Token indices sequence length is longer than the specified maximum sequence length for this model (521 > 512). Running this sequence through the model will result in indexing errors
2023-08-11 13:39:19.772 | DEBUG    | test_scrub:test_emr_image:145 - ['NAME_GIVEN', 'NAME_FAMILY', 'PHONE_NUMBER']
============================================================================================= warnings summary ============================================================================================== 
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pycountry\__init__.py:10
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pycountry\__init__.py:10: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/openadapt/test_scrub.py::test_emr_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\pytesseract\pytesseract.py:382: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return LooseVersion(

tests/openadapt/test_scrub.py::test_emr_image
tests/openadapt/test_scrub.py::test_scrub_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-DSRh12US-py3.10\lib\site-packages\setuptools\_distutils\version.py:345: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================== short test summary info ========================================================================================== 
FAILED tests/openadapt/test_scrub.py::test_emr_image - AssertionError: assert not ['NAME_GIVEN', 'NAME_FAMILY', 'PHONE_NUMBER']
================================================================================= 1 failed, 25 passed, 6 warnings in 17.88s ================================================================================= 
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\test_other\OpenAdapt> 
```

**Comment by KrishPatel13** [comment]
A Sample of Cape Results:

For reference:

Image:

image

Cape Privacy Analysis:

{'content': 'Nn\n\nFile Scheduler Encounter e-Prescription Billing Admin Master Help\n1 ae oe [NAME_GIVEN_1]\n[NAME_FAMILY_1],\n[OCCUPATION_1] Orders | Past Encounters Problem List | CompPhyOrders 4\nAll Reports vitals | Chief Complaint. ]ROS | Patient Hx | [CONDITION_1] | Medication | [MEDICAL_PROCESS_1] —_] Diagnosis\nLake encounter :06/28/2011 to 03/02/2011 +\nVitals\n[NAME_GIVEN_2] [NAME_GIVEN_3]-\nFacesheet Encounter Date|Temperature| BP BUT Bsa |Puise| Weight |Height\n[AGE_1] years\n06/26/2011 | 98. 6F sz0/e0 |29.12 as1 fo as [6s\na3/1s/2o11 |r a30/s0 |25.08 1.49 fas |iza_ |se\n1 6/25/1971 03/15/2011 | 98.7F 165/120/34.64 1.59 |76 f1as— |5a.25\n03/15/2011 | 96F s30/70 [oO a a fo a\n"[GENDER_SEXUALITY_1]\n1 OOK 1518 ALlersy\n| 9612225555 Encounter Date| Type Alleroy Severity | Reaction | Notes\n03/15/2011 [Plants [Pollen (Gra |mila Itching: sk\n4\nBLUE CROSS BLUE\nDiagnosis\nEncounter Date Type Tedscode| Problem Notes\n| e2er2011 [TIME_1] 06/28/2011 HEARTBURN [MONEY_1] | HEARTBURN\n06/28/2011 ROUTINE MEDICAL E |[MONEY_2] ROUTINE GENE\n03/15/2011 [MEDICAL_PROCESS_2] BACTE |mwmgmem [NEED FOR PRO\nFollow Up Office 03/15/2011 OTHER PROTOZOAL 1 [[MONEY_3] OTHER PROTOZ\n03/03/2011 STREPTOCOCCAL SOR [MONEY_4] |STREPTOcOcca\n03/02/2011 LUMP OR MASS IN B [[MONEY_5] [LUMP OR MASS | possible fibroa\n750 J\nBack Ache\nMedications\nTl JIpate started| Generic | Brand |strength | dose |Route|Frequency|Notes\n=] |[DRUG_1] |[DRUG_2] |[DRUG_3] |2so 1 by m [BID\n05/18/2011 [[DRUG_4] 1 |L[DRUG_5]itor |10 1 by m |DATLY\nApnly Stas 03/07/2011 |clerithro |ciarithr |2s0 1 by m [BID\n02/26/2011 |[DRUG_6] |sy[DRUG_7] |25 0.5/na1 |by m |DaILY\n02/24/2011 [[DRUG_8] |Sy[DRUG_7] |[DOSE_1]\n\nPast Illness\n\n', 'entities': [{'processed_text': 'NAME_GIVEN_1', 'text': 'le', 'best_label': 'NAME_GIVEN'}, {'processed_text': 'NAME_FAMILY_1', 'text': 'smith', 'best_label': 'NAME_FAMILY'}, {'processed_text': 'OCCUPATION_1', 'text': 'Physician', 'best_label': 'OCCUPATION'}, {'processed_text': 'CONDITION_1', 'text': 'Allergy', 'best_label': 'CONDITION'}, {'processed_text': 'MEDICAL_PROCESS_1', 'text': 'Physical Exam', 'best_label': 'MEDICAL_PROCESS'}, {'processed_text': 'NAME_GIVEN_2', 'text': 'SmithAnn', 'best_label': 'NAME_GIVEN'}, {'processed_text': 'NAME_GIVEN_3', 'text': 'R', 'best_label': 'NAME_GIVEN'}, {'processed_text': 'AGE_1', 'text': '40', 'best_label': 'AGE'}, {'processed_text': 'GENDER_SEXUALITY_1', 'text': 'Female', 'best_label': 'GENDER_SEXUALITY'}, {'processed_text': 'TIME_1', 'text': '11:38', 'best_label': 'TIME'}, {'processed_text': 'MONEY_1', 'text': '787.10', 'best_label': 'MONEY'}, {'processed_text': 'MONEY_2', 'text': 'V70.00', 'best_label': 'MONEY'}, {'processed_text': 'MEDICAL_PROCESS_2', 'text': 'VACCINATION', 'best_label': 'MEDICAL_PROCESS'}, {'processed_text': 'MONEY_3', 'text': '007.00', 'best_label': 'MONEY'}, {'processed_text': 'MONEY_4', 'text': '034,00', 'best_label': 'MONEY'}, {'processed_text': 'MONEY_5', 'text': '611.72', 'best_label': 'MONEY'}, {'processed_text': 'DRUG_1', 'text': 'Josvtev2za11', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_2', 'text': 'amoxici11', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_3', 'text': 'amoxicit', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_4', 'text': 'Lipitor', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_5', 'text': 'ip', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_6', 'text': 'Synthroia', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_7', 'text': 'ntnroi', 'best_label': 'DRUG'}, {'processed_text': 'DRUG_8', 'text': 'Synthroid', 'best_label': 'DRUG'}, {'processed_text': 'DOSE_1', 'text': '25 1 oral |DAILY', 'best_label': 'DOSE'}]}

**Comment by KrishPatel13** [comment]
here is a full list of Entities identified by Cape:

image

File: tests/openadapt/test_scrub.py [hunk 5]
```diff
@@ -19,16 +23,137 @@ def _hex_to_rgb(hex_color: int) -> tuple[int, int, int]:
...
+
+    if (
+        "entities" not in cape_response.keys()
+        or "content" not in cape_response.keys()
+        or cape_response is None
+    ):
+        pytest.skip(reason="Cape failed to return entities and content")
+
+    detect_entities = cape_response.get("entities")
+    important_entities = [
```
Comment (KrishPatel13) [inline tests/openadapt/test_scrub.py:123]: Important Entities (from Cape) are those which are similar to those which are detected by Presidio.

File: pyproject.toml [hunk 6]
```diff
@@ -83,9 +83,10 @@ pympler = "^1.0.1"
 psutil = "^5.9.5"
 pyobjc-framework-avfoundation = {version = "^9.2", markers = "sys_platform == 'darwin'"}
 fastapi = "0.98.0"
+presidio-image-redactor = "^0.0.48"
 oa-pynput = "^1.7.7"
 oa-atomacos = {version = "3.2.0", markers = "sys_platform == 'darwin'"}
-presidio-image-redactor = "^0.0.48"
+opencv-python = "^4.8.0.76"
```
Comment (KrishPatel13) [inline pyproject.toml:89]: Add opencv-python for using cv2 in test_scrub.py

**Pull request closed without merging**